tests/quic_p2p.rs

```diff
@@ -14,16 +14,60 @@ fn wait_till_connected(ev_rx: mpmc::Receiver<Event>) -> Peer {
     panic!("Didn't receive the expected ConnectedTo event");
 }
 
```

Comment:
This seems duplicate to the above code in test_utils? Is this perhaps an indication of some code/tests being in the wrong place?

Reply:
Some are integration and other unit tests, and other examples.
The problem is integration tests or example can only use what is `pub` from the library code.

Reply:
Maybe it's worth exposing `EventReceiever` as a helper struct then? As it's used in almost the same way in examples, integration tests, and unit tests, moving it to `test_utils` and making it public might be a good way to handle that.

Reply:
I don't think so, though this can always be updated later. But essentially we provide 2 channels so 2 different consumer can listen for the different events. The reason this struct make sense is in the existing tests we want to listen to both channels in the same place.